include/matx/core/nvtx.h

```diff
@@ -0,0 +1,275 @@
+                                             NVTX_YELLOW,
+                                             NVTX_TEAL,
+                                             NVTX_PINK,
+                                             NVTX_WHITE
+                                             };
+
+inline uint64_t curColorIdx;
+inline std::map< int, nvtxRangeId_t>  eventMap;
+
+inline matx_nvxtLogLevels globalNvtxLevel = matx_nvxtLogLevels::MATX_NVTX_LOG_API;
+
+//////  macros to ensure custom variable names for every call  ////////
+#define MATX_CONCAT(a, b) MATX_CONCAT_INNER(a, b)
+#define MATX_CONCAT_INNER(a, b) a ## b
+
+#define MATX_UNIQUE_NAME(base) MATX_CONCAT(base, __COUNTER__)
+///////////////////////////////////////////////////////////////////////
+
+
+////////////             Enable NVTX Macros          /////////////////
+
+#ifdef MATX_NVTX_FLAGS
+
+  ///\todo update to use C++20 runtime fucntion for actual call location
+  /// https://en.cppreference.com/w/cpp/utility/source_location
+  #define MATX_NVTX_1( message ) matx::NvtxEvent MATX_UNIQUE_NAME(nvtxFlag_)( __FUNCTION__, message );
+  #define MATX_NVTX_2( message, nvtxLevel ) matx::NvtxEvent MATX_UNIQUE_NAME(nvtxFlag_)( __FUNCTION__, message, nvtxLevel );
+
+  #define MATX_NVTX_X(x,A,B,FUNC, ...)  FUNC
+
+  // The macro that the programmer uses
+  #define MATX_NVTX_START(...)  MATX_NVTX_X(,##__VA_ARGS__,\
+                                MATX_NVTX_2(__VA_ARGS__),\
+                                MATX_NVTX_1(__VA_ARGS__)\
+                                )
+
+  #define MATX_NVTX_START_RANGE( message, nvtxLevel, id ) matx::NvtxEvent MATX_UNIQUE_NAME(nvtxFlag_)( __FUNCTION__, message, nvtxLevel, id );
+
+  #define MATX_NVTX_END_RANGE( id ) endEvent( id );
+  
+  #define MATX_NVTX_SET_LOG_LEVEL( nvtxLevel ) setNVTXLogLevel( nvtxLevel );
+  
+////////////             Disable NVTX Macros          /////////////////
+
+#else
+
+  #define MATX_NVTX_1( message );
+  #define MATX_NVTX_2( message, customId );
+  #define MATX_NVTX_START_RANGE( message, nvtxLevel, id );
+
+  #define MATX_NVTX_X(x,A,B,FUNC, ...)  FUNC
+
+  // The macro that the programmer uses
+  #define MATX_NVTX_START(...)    MATX_NVTX_X(,##__VA_ARGS__,\
+                                  MATX_NVTX_2(__VA_ARGS__),\
+                                  MATX_NVTX_1(__VA_ARGS__)\
+                                  )
+                                  
+  #define MATX_NVTX_END_RANGE( id );
+  
+  #define MATX_NVTX_SET_LOG_LEVEL( nvtxLevel );
+
+#endif
+////////////////////////////////////////////////////////////////////////////////
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+///
+///\brief Utility Function to set Global Log Level. should be called through the 
+///       MATX_NVTX_SET_LOG_LEVEL macro with the same parameters
+///
+////////////////////////////////////////////////////////////////////////////////
+[[maybe_unused]] static void setNVTXLogLevel( matx_nvxtLogLevels newNVTXLevel)
```

Comment:
is this attribute required if you're not calling the function? I thought it was only required on parameters